### ISSUE TITLE
Новое свойство: числовой диапазон

### ISSUE THREAD

**Issue by facepoint** [instantsoft/icms2#133]
Столкнулся при формировании каталога с еще одной проблемой. Товар отличается только одним параметром, - разным диаметром от 10 до 50. Было бы мегаудобно, если было бы соответствующее свойство, которое позволяло бы вбивать этот диапазон и выводить также в фильтре.

**Comment by mafru** [comment]
А разве в текущей реализации свойство выпадаюший список не справится с вашей задачей.
К примеру:
0 - 10
10 - 20
20 - 30
30 - 40
40 - 50

**Comment by vlad7090** [comment]
это надо предусмотреть ОГРОМное количество таких значений.

а так задал при добавлении товара от 100 до 300, а у доругого от 10 до 100 может быть....и таких вариаций тысячи.

А человек в фильтре вбив например 90-95 сможет найти те товары, которые входят в этот диапазон

**Comment by EnotAlex** [comment]
аналогичная ситуация и с полем поиск **по году** по диапазону **от** и **до**;
всяко удобней для ленивого пользователя выбирать год из 2-ух выпадающих списков, чем тыкать цифры на клаве, а вот сам выпадающий список берётся из **Предустановленные значения**, где мы уже сами перечисляем нужные нам года.
посмотреть пример можно тут https://auto.ru/cars/ford/all/?listing=listing&sort_offers=cr_date-DESC&top_days=off&currency=RUR&output_type=list&year_from=1982&year_to=2016&beaten=1&customs_state=1&page_num_offers=1

**Comment by vlad7090** [comment]
видимо самый логичный способ решить этот вопрос со свойствами, сделать их такими же расширяемыми как и поля типов контента

**Comment by fuzegit** [comment]
Я может чего не понимаю, но фильтр по диапазону есть.

![Поле](http://storage1.static.itmages.com/i/16/0306/h_1457295600_2576245_e6783922db.png)

![Свойство](http://storage7.static.itmages.com/i/16/0306/h_1457295641_5772132_7589534393.png)

**Comment by Soul-git** [comment]
"Я может чего не понимаю, но фильтр по диапазону есть"
На сколько я понял, народу нужен диапазон числовых значений для выбора пользователем, а не их фильтрация. Сейчас попробую "накидать" (где-то начинал).

**Comment by Soul-git** [comment]
[http://www.instantcms.ru/users/files/download5865.html](url)
Что- то вроде этого? (тестовый вариант для ознакомления - не доработан)

**Comment by EnotAlex** [comment]
с этими настройками выходит только так 
![scrin_god_ot_do_1_cr](https://cloud.githubusercontent.com/assets/17655779/13557534/173f9874-e403-11e5-9b43-38ddf091d739.png)

**Comment by EnotAlex** [comment]
всё что нужно, это взять данные из поля **Предустановленные значения**
![scrin_god_ot_do_2_cr](https://cloud.githubusercontent.com/assets/17655779/13557557/ca912168-e403-11e5-8c8e-01ef0b657705.png)
 и поместить их в 2-а выпадающих списка, т.е. получится вот так 
![ot_do_god_cr](https://cloud.githubusercontent.com/assets/17655779/13557570/09b83b6a-e404-11e5-9b3e-a1021de481c1.png)

**Comment by vlad7090** [comment]
в числе можно задать только одно число. но допустим характеристика товара Ширина. И может иметь значение от 100 до 500.
Если подойти решениями имеющихся свойств, то надо сделать список предустановленых значений - 400 цифр от 100 до 500 и далее выбирать их все.

В случае с числовым диапозоном, достаточно будет задать верхнюю и нижнюю границу.

**Comment by vlad7090** [comment]
и не знаю на сколько это сложно, но в имеющемся числовом диапазоне было бы отлично иметь ползунки на аяксе (опционально) - не критичное дополнение конечно, можно и без него прекрасно работать с фильтрамиЭ, но как доп.фишка хорошее дополнение

**Comment by Soul-git** [comment]
посмотрите у меня в файлах в профиле, есть "ползунок" (range.zip), правда не доработан - еще "не отлажен" функционал фильтрации по диапазону и значениям.

**Comment by Val-Git** [comment]
> но допустим характеристика товара Ширина. И может иметь значение от 100 до 500.
@vlad7090, какой например товар может иметь ширину от 100 до 500?

**Comment by pin-git** [comment]
@Val-Git товар А имеет ширину 300 мм, товар Б - 300 см, товар В - 300 дм, как понял из выше поданного суть не в товаре.

**Comment by Val-Git** [comment]
@pin-git, не вникал глубоко в суть проблемы, но думаю задача ТС решается установкой двух полей числового типа (если хочется через диапазон). Хотя, IMHO, указываемый параметр имеет один размер и логичнее завести несколько одинаковых записей с различным значением параметра. Или же задать изменяемый параметр в выпадающем списке с несколькими значениями

**Comment by vlad7090** [comment]
например толщина плиты может быть любой начиная от 20 мм до 40мм. проставлять 20 значений не логично, а вот задать сразу диапазон самое то.
при этом у другого товара толщина уже от 100мм до 120мм. 
Если бы было свойство "диапазон чисел", то это свойство можно было бы применить сразу ко всем категориям товаров.

**Comment by vlad7090** [comment]
Soul-git не нашел где смотреть:(

**Comment by Soul-git** [comment]
vlad7090 посмотрите в файлах моего профиля на instantcms.ru . Но это лишь тестовый вариант и использовать его без доработки на проекте нельзя.

**Comment by iton1** [comment]
Очень нужно поле диапазон чисел! 

Сейчас есть поле число, но при добавление вписывается только одно число, а нужно два, и чтобы по этому диапазону в фильтре можно было искать.

Приведу несколько примеров:

1) Новостройки. 

Добавляем новостройку на сайт, у новостроек есть к примеру такие данные как "цена" и "площадь в диапазоне" которые нам надо добавить на сайт
[http://joxi.ru/1A5xzqWuKEpy02.jpg](http://joxi.ru/1A5xzqWuKEpy02.jpg) и по которым в поиске в фильтре могли бы ее найти. 
Но добавить два числа в поле у нас нет возможности тк можно добавить можно только одну цифру, примерно как это выглядело если было поле диапазон чисел  [http://joxi.ru/J2bV103s4XOex2](http://joxi.ru/J2bV103s4XOex2) ) 

2)  Пример на кредитах.
Добавляя кредит на сайт у него есть значение "Сумма" которая указана в диапазоне чисел
[http://joxi.ru/1A5xzqWuKny8q2](http://joxi.ru/1A5xzqWuKny8q2) , но добавить ее на сайт и искать в фильтре не получиться.

**Comment by EnotAlex** [comment]
поддерживаю

**Comment by lokanaft** [comment]
Как связаны новостройки, кредиты и "Система управления сообществами"?

**Comment by Val-Git** [comment]
@lokanaft поддерживаю =)

**Comment by iton1** [comment]
Это как пример для наглядности как это поле можно использовать на сайтах, у всех тематики сайтов разные(авто, кредиты, новостройки, стоматологи и тд).

**Comment by globator88** [comment]
Столкнулись мы подобной проблемой. Сделали тип поля "От и До" http://joxi.ru/52aYZJBfGN7172 Но не для свойств, а в вкладке поля. Нужно было реализовать диапазон зарплаты от и до, и возраст. Но в фильтре пока не работает (пока лен реализовать), так как данные добавляется в одну таблицу.

**Comment by vlad7090** [comment]
globator88, можете поделиться наработкой?

**Comment by iAndroS** [comment]
Не всегда загодя известно "От и До"...
Думаю, надо просто делать два РАЗНЫХ числовых поля (пусть даже теже "от" и "до") и какой-то обработчик (поле???) уже выводить, их сравнивающий и выводящий соответствующую форму - тогда и предустановленных значений не нужно будет, и данные четко прописаны в нужных полях контента.
Думаю, Локанафту это будет под силу после авипарса )))

**Comment by EnotAlex** [comment]
получится вроди этого 
![scrin_god_ot_do_4](https://cloud.githubusercontent.com/assets/17655779/13725322/1d4e1712-e8af-11e5-8182-cb8e79e82aef.png)
![ot_do_god_4](https://cloud.githubusercontent.com/assets/17655779/13725323/21d4a9c2-e8af-11e5-9d7e-a8407922050f.png)

**Comment by iAndroS** [comment]
Да, на выходе так. Но на входе предустанавливать ничего не надо будет - и это здорово!

**Comment by globator88** [comment]
> globator88, можете поделиться наработкой?

Пока не смогу

> получится вроди этого:

А что будет если значение будет намного больше? На пример зарплата от 100 000 до 200 000р
Логично сделать как тип поля "Число"

**Comment by EnotAlex** [comment]
В общем потом каждый сам решает, как он будет это использовать для своего проекта.

**Comment by EnotAlex** [comment]
Для зарплат, кто-нибудь придумает другое поле или будет пользоваться этим 
![scrin_god_ot_do_1_cr2](https://cloud.githubusercontent.com/assets/17655779/13725425/4e9da8de-e8b2-11e5-8a99-c525dcfa105d.png)
